add imaps STARTTLS support

WnP opened the pull request:
This pull request add imap connection using [STARTTLS][dw], which as you can read is different from a simple SSL connection

Use `imaps` as protocol for this feature

[dw]: http://wiki.dovecot.org/SSL "Dovecot wiki SSL"

coddingtonbear commented:
Awesome; thanks for the patch, @WnP!  Would you mind adding this to the docs -- https://github.com/coddingtonbear/django-mailbox/blob/master/docs/topics/mailbox_types.rst, and would you be deeply offended if I were to alter this such that it wasn't `imaps` but `imap+tls` so this might be a little more consistent with the way existing "transports" are implemented?

WnP commented:
@coddingtonbear no problem, I guess `+tls` is a better option too, also I was thinking about the documentation too, but english is not my mother-tong, but I can do my best if you want

coddingtonbear commented:
Ahh, I understand -- I hadn't guessed that you weren't a native speaker from your original post above; I'd be glad to write the docs for you if you're not comfortable.  Thanks again, @WnP!

WnP commented:
@coddingtonbear I've refactored the feature to use `+tls` and add some documentation, I've also rebased this branch to squash useless commit for the history, let me know your through

coddingtonbear commented:
Thanks for the patch @Wnp, and sorry for the delay in reviewing it!  This will be released as part of 4.5.0.

eriktelepovsky commented:
This solution won't work with Python 2.7:

https://docs.python.org/3/library/imaplib.html#imaplib.IMAP4.starttls

https://stackoverflow.com/questions/28214673/connect-to-imap-exchange-with-starttls-in-python-2-7-not-3
